src/figuresClasses.ts

```diff
@@ -1,11 +1,76 @@
-export interface Figure {}
 
-export class Triangle implements Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a = 0,
+    public b = 0,
+    public c = 0,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides of a triangle must be greater than 0');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('must be less than the sum of the other two sides');
+    }
+}
+
+getArea(): number {
+  const s = (this.a + this.b + this.c) / 2;
+
+  const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+  return Math.floor(area * 100) / 100;
+}
+}
+
+export class Circle implements Figure {
+  public shape: Shape = 'circle';
+  constructor(
+    public color: Color,
+    public radius: number = 0,
+  ) {
+    if (radius <= 0) {
+      throw new Error(`it can't be a circle`);
```

Comment:
The error message is vague. Consider changing it to something more descriptive, like: 'Radius must be greater than 0 to form a valid circle.'